src/components/grid-infinite-scroll/grid-infinite-scroll.tsx

```diff
@@ -0,0 +1,253 @@
+import {
+  Component,
+  ComponentInterface,
+  Element,
+  Event,
+  EventEmitter,
+  EventListenerEnable,
+  Listen,
+  Method,
+  Prop,
+  QueueApi,
+  State,
+  Watch
+} from "@stencil/core";
+
+@Component({
+  shadow: false,
+  styleUrl: "grid-infinite-scroll.scss",
+  tag: "gx-grid-infinite-scroll"
+})
+export class GridInfiniteScroll implements ComponentInterface {
+  private thrPx = 0;
+  private thrPc = 0;
+  private scrollEl?: HTMLElement;
+  private didFire = false;
+  private isBusy = false;
+
+  @Element() el!: HTMLElement;
+  @State() isLoading = false;
+
+  /**
+   * Query selector where the infinitie scroll would be listening to scroll events.
+   */
+  @Prop() infiniteScrollContainer = "div";
+
+  /**
+   * This property must be bounded to grid item count property.
+   * It's unique purpose is to trigger gxInfinite as many times as needed to fullfill the Container space when the intial batch does not overflow the main container
+   */
+  @Prop() itemCount = 0;
+
+  @Prop({ context: "queue" })
+  queue!: QueueApi;
+  @Prop({ context: "enableListener" })
+  enableListener!: EventListenerEnable;
+
+  /**
+   * The threshold distance from the bottom
+   * of the content to call the `infinite` output event when scrolled.
+   * The threshold value can be either a percent, or
+   * in pixels. For example, use the value of `10%` for the `infinite`
+   * output event to get called when the user has scrolled 10%
+   * from the bottom of the page. Use the value `100px` when the
+   * scroll is within 100 pixels from the bottom of the page.
+   */
+  @Prop() threshold = "15%";
+
+  /**
+   * If `true`, the infinite scroll will be hidden and scroll event listeners
+   * will be removed.
+   *
+   * Set this to true to disable the infinite scroll from actively
+   * trying to receive new data while scrolling. This is useful
+   * when it is known that there is no more data that can be added, and
+   * the infinite scroll is no longer needed.
+   */
+  @Prop() disabled = false;
+
+  /**
+   * The position of the infinite scroll element.
+   * The value can be either `top` or `bottom`.
+   */
+  @Prop() position: "top" | "bottom" = "bottom";
+
+  /**
+   * Emitted when the scroll reaches
+   * the threshold distance. From within your infinite handler,
+   * you must call the infinite scroll's `complete()` method when
+   * your async operation has completed.
+   */
+  @Event() gxInfinite!: EventEmitter<void>;
+
+  componentWillLoad() {
+    this.itemCountChanged();
+  }
+
+  @Watch("itemCount")
+  public itemCountChanged() {
+    setTimeout(() => {
```

Comment:
Not sure why this is deferred? Have you considered using `request​Animation​Frame` instead?